src/numerics/CVodesIntegrator.cpp

```diff
@@ -513,18 +546,27 @@ void CVodesIntegrator::integrate(double tout)
                 "Components with largest weighted error estimates:\n{}",
                 flag, m_error_message, f_errs, getErrorInfo(10));
         }
+        if (flag == CV_ROOT_RETURN) {
+            // Stop early at root (e.g., advance limit reached)
+            root_triggered = true;
+            break;
+        }
         nsteps++;
     }
-    int flag = CVodeGetDky(m_cvode_mem, tout, 0, m_y);
+    // Align solution time to the requested tout when no root event occurred; this
+    // avoids small positive overshoots from CV_ONE_STEP, which can otherwise cause
+    // callers to see non-monotonic target times.
+    double t_eval = root_triggered ? m_tInteg : tout;
+    int flag = CVodeGetDky(m_cvode_mem, t_eval, 0, m_y);
```

Comment:
There's really no expectation for CVODES to stop at the specified `tout`. My take on this would be:
```suggestion
    // Interpolate the solution to either the user-specified output time or
    // the time at which a root event occurred.
    double t_eval = root_triggered ? m_tInteg : tout;
    int flag = CVodeGetDky(m_cvode_mem, t_eval, 0, m_y);
```